### PR TITLE
Adapt for first use

### PR DESCRIPTION
**Pull request by Screenfeed**
This PR brings various improvements after trying to use this project for PLL Pro's tests.

## WP's test suite location

WP's test suite is not installed in this project anymore, but in the client's `tmp` folder instead. This prevents to have to:
- re-install the suite each time this project is updated,
- search deep inside the `vendor` folder when we need to access WP's files.

## Folders

- A `$rootDir` has been added to `WP_Syntex\Polylang_Phpunit\Bootstrap::__construct()` (and to the `bootstrapSuite()` functions). This allows more flexibility, the "tests dir" not being forced to be 1 level down the "root dir" (project dir) anymore (see the constants `WPSYNTEX_PROJECT_PATH` and `WPSYNTEX_TESTS_PATH`).
- New constant `WPSYNTEX_TESTSUITE_PATH` that contains the path to the test suite: `tmp/wordpress-tests-lib`. It is used in 1 test class in PLL Pro (`PLL_Translation_Walker_Blocks_Test`).

## Tools

- A 3rd parameter has been added to `WP_Syntex\Polylang_Phpunit\TestCaseTrait::getTestData()` to replicate what is already done in PLL Pro. It allows to select only a subset instead of the entire data set.

## Bash scripts

- Fix a few things.
- New file `colors.sh` containing color definitions for the messages.
- Harmonized messages and added some colors to error and success messages.

## PHPStan

- A class `DummyTestCase` has been added so PHPStan can cover `WP_Syntex\Polylang_Phpunit\Integration\TestCaseTrait`.
- A `(array)` cast has been added in `WP_Syntex\Polylang_Phpunit\Bootstrap::isGroup()`.
- Some constants have been added to PHPStan's ignore list.

## PHP 7.0

Bumped from php 5.6 to 7.0.

## Readme

We make typo.

### CODE REVIEW

File: bin/distribute.sh [hunk 1]
```diff
@@ -4,22 +4,36 @@
...
 else
-	local PACKAGE_NAME=$(basename -s .git `git config --get remote.origin.url`)
+	PACKAGE_NAME=$(basename -s .git `git config --get remote.origin.url`)
 fi
 
 if [[ '--no-npm' != $2 ]]; then
 	echo "Running build..."
-	npm install && npm run build # minify js and css files.
+	# minify js and css files.
+	npm install && npm run build
```
Comment (Hug0-Drelon) [inline bin/distribute.sh:31]:
```suggestion
	npm install --package-lock-only && npm ci && npm run build
```
This way `package-lock.json` is all updated and pretty and `node_modules` are refreshed.

File: composer.json [hunk 2]
```diff
@@ -25,6 +28,11 @@
 			"WP_Syntex\\Polylang_Phpunit\\": "UnitTests/"
 		}
 	},
+	"autoload-dev": {
+		"psr-4": {
+			"WP_Syntex\\Polylang_Phpunit\\Fixtures\\": "Fixtures/"
```
Comment (Hug0-Drelon) [inline composer.json:33]: I didn't pay attention to it in the other files, but why not camelcase here?

File: phpstan.neon.dist
Comment (Hug0-Drelon) [inline phpstan.neon.dist:1]: Wouldn't it be better to have a `bootstrap.php` file for PHPStan where we could declare our constants (and also be able to do other stuff)?
See https://phpstan.org/config-reference#bootstrap

File: composer.json (hunk 2 above)
Comment (Screenfeed) [reply composer.json:33]: I don't remember the exact reason but we reused the syntax defined 2 years ago in https://github.com/polylang/polylang-pro/pull/990 (`WP_Syntex\Polylang_Pro`).

File: phpstan.neon.dist
Comment (Screenfeed) [reply phpstan.neon.dist:1]: You mean to prevent this?
```
	ignoreErrors:
		- '#^Constant WPSYNTEX_PROJECT_PATH not found\.$#'
		- '#^Constant WPSYNTEX_FIXTURES_PATH not found\.$#'
		- '#^Constant WPSYNTEX_TESTS_PATH not found\.$#'
```

File: composer.json (hunk 2 above)
Comment (Hug0-Drelon) [reply composer.json:33]: Let's keep it that way then 👍

File: phpstan.neon.dist
Comment (Screenfeed) [reply phpstan.neon.dist:1]: Fantastic, it works 👍 
[fd46a89](https://github.com/polylang/wp-phpunit/pull/9/commits/fd46a89dc430c961794ee522ac9c6145a21602a6)

File: phpstan-bootstrap.php
Comment (Hug0-Drelon) [inline phpstan-bootstrap.php:1]: It misses an empty line at the end, otherwise 👍

Comment (Screenfeed) [reply phpstan-bootstrap.php:1]: Weird, I have the empty line locally :|

Comment (Hug0-Drelon) [reply phpstan-bootstrap.php:1]: 🪄

Comment (Hug0-Drelon) [reply phpstan-bootstrap.php:1]: Nevermind then

**Comment by Screenfeed** [comment]
Funny that Travis is not running here btw 😅

**Comment by Hug0-Drelon** [comment]
> Funny that Travis is not running on this repository btw 😅

There's no `travis.yml` 😝

**Review by Hug0-Drelon — APPROVED** [review]

**Comment by Screenfeed** [comment]
> There's no `travis.yml` 😝

I know! 🥲

File: phpstan-bootstrap.php [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<?php
+/**
+ * Bootstraps PHPStan
+ * php version 5.6
```
Comment (Chouby) [inline phpstan-bootstrap.php:4]: I guess we can forget PHP 5.6.

That said, is this file necessary? https://phpstan.org/user-guide/discovering-symbols#global-constants

File: bin/build.sh [hunk 4]
```diff
@@ -4,22 +4,25 @@
...
 # $2 string '--no-npm' to not run npm.
 
+echo "Installing PHP packages..."
+
+# Include color values (must be done before `rm -rf vendor`).
+PARENT_DIR=$( cd "$(dirname "${BASH_SOURCE[0]}")" ; pwd -P )
+. "$PARENT_DIR/colors.sh"
+
+# Make sure to remove all traces of development dependencies.
+rm -rf vendor
```
[the remark this answers is not in the document]
Comment (Screenfeed) [reply bin/build.sh:14]: I'm not the one adding this, I only moved it from the line 14.

File: UnitTests/Bootstrap.php [hunk 5]
```diff
@@ -90,18 +99,17 @@ public function initTestSuite() {
...
-			 * Unit tests:
-			 * Load Patchwork before everything else in order to allow us to redefine WordPress, 3rd party, and plugin's
-			 * functions.
-			 */
-			$patchworkPath = __DIR__ . '/../vendor/antecedent/patchwork/Patchwork.php';
+		/**
+		 * Load Patchwork before everything else in order to allow us to redefine WordPress, 3rd party, and plugin's
+		 * functions.
+		 */
+		$patchworkPath = __DIR__ . '/../vendor/antecedent/patchwork/Patchwork.php';
```
[the remark this answers is not in the document]
Comment (Screenfeed) [reply UnitTests/Bootstrap.php]: Nice catch 👍

File: phpstan-bootstrap.php (hunk 3 above)
Comment (Screenfeed) [reply phpstan-bootstrap.php]: Bumped to php 7.0 in [75cdb0f](https://github.com/polylang/wp-phpunit/pull/9/commits/75cdb0fd662afd689ec70e96a798e683e7fb7788).

File: Fixtures/DummyTestCase.php [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+class DummyTestCase extends WP_UnitTestCase {
+
+	/**
+	 * List of active plugins.
+	 *
+	 * @var array<string>
+	 */
+	protected $activePlugins = [
+		'wp-all-import-pro/wp-all-import-pro.php',
+	];
```
[the remark this answers is not in the document]
Comment (Screenfeed) [reply Fixtures/DummyTestCase.php]: Done in [c9b1865](https://github.com/polylang/wp-phpunit/pull/9/commits/c9b1865fefc23948a2157a5d47192d8fc9e4b4fc).
I had to use Yoast's `setUp()` method instead of the "native" `set_up()` method because of PHPCS' snakeCase sniff, but we could have used `set_up()`.

[the remark this answers is not in the document]
Comment (Chouby) [reply Fixtures/DummyTestCase.php]: Do we really need `setUp()` in this dummy class now that the property is defined in the trait? The goal of my proposal was in fact to try to remove this dummy plugin in this class.

File: phpstan-bootstrap.php (hunk 3 above)
Comment (Chouby) [reply phpstan-bootstrap.php]: Strange. As far as I remember, we don't have such need in Polylang. Would this help? https://phpstan.org/config-reference#constants

File: Fixtures/DummyTestCase.php (hunk 6 above)
[the remark this answers is not in the document]
Comment (Screenfeed) [reply Fixtures/DummyTestCase.php]: Removed in [617439e](https://github.com/polylang/wp-phpunit/pull/9/commits/617439ea2d7f0682f97d5bfba4c7053969f62628).

**Comment by Chouby** [comment]
Hidden files and folders are added in the zip file when running `composer dist`. Could you remove them?

**Comment by Screenfeed** [comment]
> Hidden files and folders are added in the zip file when running `composer dist`. Could you remove them?

There is no `composer dist` command here, I guess you're talking about PLL Pro?
